core/src/main/kotlin/xtdb/vector/IVectorIndirection.kt

```diff
@@ -0,0 +1,37 @@
+package xtdb.vector
+
+interface IVectorIndirection {
+    fun valueCount(): Int
+
+    fun getIndex(idx: Int): Int
+
+    operator fun get(idx: Int): Int {
+        return getIndex(idx)
+    }
+
+    @Suppress("DEPRECATION")
```

Comment:
what's the deprecation here, OOI? can we avoid it?

Reply:
It's on the `formatted` fn below.